core/src/main/scala/org/apache/spark/BarrierCoordinator.scala

```diff
@@ -199,13 +245,12 @@ private[spark] class BarrierCoordinator(
   }
 
   override def receiveAndReply(context: RpcCallContext): PartialFunction[Any, Unit] = {
-    case request @ RequestToSync(numTasks, stageId, stageAttemptId, _, _) =>
+    case request: RequestToSync =>
```

Comment:
This is not necessary, please revert this change.

Reply:
@jiangxb1987 `RequestToSync` is no a sealed trait which doesn't seem to allow for variable extraction.

Reply:
oh i see, sound good!